### ISSUE TITLE
`cyclonedx-py` fails with `poetry` v1.3 due to new v2 lock format

### ISSUE THREAD

**Issue by tewfik-ghariani** [CycloneDX/cyclonedx-python#467]
Starting `poetry` v1.3, the lock file format has changed essentially in the part of specifying the files/hashes per package instead of a dedicated `metadata.files`

_lock v1_

<img width="1344" alt="Screenshot 2022-12-12 at 19 06 00" src="https://user-images.githubusercontent.com/11073604/207120934-265e3c9d-3ad6-4b2d-a6f1-59f07c4adb9d.png">

_lock v2_

<img width="1344" alt="Screenshot 2022-12-12 at 19 09 15" src="https://user-images.githubusercontent.com/11073604/207121612-b790805d-50cb-4607-9283-ff307f9587f5.png">

Refs: https://github.com/python-poetry/poetry/pull/6393

As a result, the `cyclonedx-py` tool is no longer able to generate the bom file because it cannot parse the new format

```
$ cyclonedx-py --poetry -o target/bom.xml
Traceback (most recent call last):
  File "/usr/local/bin/cyclonedx-py", line 8, in <module>
    sys.exit(main())
  File "/usr/local/libexec/cyclonedx-bom/lib/python3.9/site-packages/cyclonedx_py/client.py", line 307, in main
    CycloneDxCmd(args).execute()
  File "/usr/local/libexec/cyclonedx-bom/lib/python3.9/site-packages/cyclonedx_py/client.py", line 147, in execute
    output = self.get_output()
  File "/usr/local/libexec/cyclonedx-bom/lib/python3.9/site-packages/cyclonedx_py/client.py", line 84, in get_output
    parser = self._get_input_parser()
  File "/usr/local/libexec/cyclonedx-bom/lib/python3.9/site-packages/cyclonedx_py/client.py", line [295](https://git.mam.dev/acdc/gocd/pipeline-templates/-/jobs/9232334#L295), in _get_input_parser
    return PoetryParser(poetry_lock_contents=input_data,
  File "/usr/local/libexec/cyclonedx-bom/lib/python3.9/site-packages/cyclonedx_py/parser/poetry.py", line 44, in __init__
    for file_metadata in poetry_lock['metadata']['files'][package['name']]:
KeyError: 'files'
```

**Comment by jkowalleck** [comment]
thanks for the report and the RP #468.
I will look into it ASAP.

**Comment by jkowalleck** [comment]
should be fixed via [3.10.0](https://github.com/CycloneDX/cyclonedx-python/releases/tag/v3.10.0).

could you test and confirm, @tewfik-ghariani ?

**Comment by tewfik-ghariani** [comment]
@jkowalleck Works like a charm, thanks a lot! 🎉
